Extension/src/LanguageServer/Providers/renameProvider.ts

```diff
@@ -3,110 +3,100 @@
+                        const metadata: vscode.WorkspaceEditEntryMetadata = {
+                            needsConfirmation: reference.type !== refs.ReferenceType.Confirmed,
+                            label: refs.getReferenceTagString(reference.type, false, true),
+                            iconPath: refs.getReferenceItemIconPath(reference.type, false)
+                        };
+                        workspaceEditResult.replace(uri, range, newName, metadata);
                     }
-                    if (referencesResult && (referencesResult.referenceInfos === null || referencesResult.referenceInfos.length === 0)) {
+
+                    if (result.referenceInfos === null || result.referenceInfos.length === 0) {
                         vscode.window.showErrorMessage(localize("unable.to.locate.selected.symbol", "A definition for the selected symbol could not be located."));
                     }
-                    resolve(workspaceEdit);
-                });
-                workspaceReferences.startRename(params);
-            };
 
-            if (DefaultClient.referencesRequestPending || workspaceReferences.symbolSearchInProgress) {
-                const cancelling: boolean = DefaultClient.referencesPendingCancellations.length > 0;
-                DefaultClient.referencesPendingCancellations.push({
-                    reject: () => {
-                        --DefaultClient.renameRequestsPending;
-                        // Complete with nothing instead of rejecting, to avoid an error message from VS Code
-                        const workspaceEdit: vscode.WorkspaceEdit = new vscode.WorkspaceEdit();
-                        resolve(workspaceEdit);
-                    }, callback
-                });
-                if (!cancelling) {
-                    workspaceReferences.referencesCanceled = true;
-                    if (!DefaultClient.referencesRequestPending) {
-                        workspaceReferences.referencesCanceledWhilePreviewing = true;
-                    }
-                    this.client.languageClient.sendNotification(CancelReferencesNotification);
+                    resolve(workspaceEditResult);
                 }
+                return;
+            };
+
+            if (requestCanceled === refs.CancellationSender.None) {
+                workspaceReferences.renamePending = true;
+                workspaceReferences.setReferencesResultsCallback(resultsCallback);
+
+                // Send the request to language server
+                const params: RenameReferencesParams = {
+                    newName: newName,
+                    position: Position.create(position.line, position.character),
+                    textDocument: { uri: document.uri.toString() }
+                };
+                workspaceReferences.startRename(params);
             } else {
-                callback();
+                // Only complete the request at this point if the request to language server
```

Comment:
Can you change all (existing) uses of `to language server` to `to the language server` and `when language` to `when the language`?

Reply:
This comment was removed, as logic changed.